packages/flutter/lib/src/rendering/editable.dart

```diff
@@ -232,6 +232,7 @@ class RenderEditable extends RenderBox with RelayoutWhenSystemFontsChangeMixin {
        assert(clipBehavior != null),
        _textPainter = TextPainter(
          text: text,
+         maxLines: maxLines,
```

Comment:
If you do this then you'll likely have to add similar logic to the maxLines setter too.

Reply:
Hmm, I've tried to apply a similar logic in [this line](https://github.com/flutter/flutter/pull/80437/files#diff-486a5b5906b709a6d958a4334055606ffb91dafcd1c384ac4245b4320817f4c8L1214) but we would need a default value for `maxLines`. Something like:

`maxLines: maxLines ?? (overflow != null ? 100 : null)` where 100 is the default max lines value.

Do you have any suggestions?

Reply:
If you pass `maxLines` here to the text painter then it probably needs to be updated in the `RenderEditable.maxLines` as well is what I meant (otherwise when you change `RenderEditable.maxLines` it doesn't update the painter as it would in the constructor).